.github/workflows/msbuild.yml

```diff
@@ -66,3 +71,65 @@ jobs:
       # This is the best way to fix an issue in master that was only caught by the nightly build.
       all_platforms: ${{ contains(github.event.pull_request.labels.*.name, 'run-deep-tests') || contains(github.event.push.labels.*.name, 'run-deep-tests')}}
       num_shards: 5
+
+  test-coverage-analysis:
+    runs-on: ubuntu-20.04
+    needs: [xunit-tests, integration-tests]
+    steps:
+      # Check out Dafny so that highlighted source is possible
+      - name: Checkout Dafny
+        uses: actions/checkout@v3
+        with:
+          path: dafny
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: ${{env.dotnet-version}}
+
+      - name: Install dotnet-reportgenerator and coverage
+        run: |
+          dotnet tool install --global dotnet-reportgenerator-globaltool
+          dotnet tool install --global dotnet-coverage
+
+      - name: Download integration test artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: integration-test-results-ubuntu-20.04
+
+      - name: Download unit test artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: unit-test-results-ubuntu-20.04
+
+      - name: Show files
+        run: ls -lR
+
+      - name: Merge reports
+        run: |
+          dotnet coverage merge DafnyLanguageServer.Test.coverage -o DafnyLanguageServer.Test.cobertura.xml -f cobertura
+          reportgenerator \
+            -reports:"./**/coverage.cobertura.xml;*.cobertura.xml" \
+            -assemblyfilters:"+AutoExtern;+dafnylanguageserver.dll;+Dafny;+DafnyCore;+DafnyDriver;+DafnyRuntime;+DafnyTestGeneration" \
```

Comment:
I think it, or something like it, is necessary. Otherwise, it includes various system assemblies in the report, dropping coverage significantly and cluttering up the report.